Amazon.IonObjectMapper/Factory/DefaultIonWriterFactory.cs

```diff
@@ -24,37 +24,18 @@ namespace Amazon.IonObjectMapper
     /// </summary>
     public class DefaultIonWriterFactory : IIonWriterFactory
     {
-        private readonly IonSerializationFormat format = TEXT;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="DefaultIonWriterFactory"/> class.
-        /// </summary>
-        public DefaultIonWriterFactory()
```

Comment:
Why removing the public interface? Is it possible to set the default values for format?

Reply:
This parameterless, empty body constructor will be provided by the compiler automatically and thus not needed in the code, see https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors#parameterless-constructors

Reply:
I'm a little surprised that we had them in the first place then. Does it still use the same default format as before? Seems we don't have a unit test for factories?

Reply:
_"If a class has no explicit instance constructors, C# provides a parameterless constructor that you can use to instantiate an instance of that class"_. Previously we have a constructor with format parameter `public DefaultIonWriterFactory(IonSerializationFormat format)`, so the parameterless constructor is also needed.

It will use the format specified in the options, which is Ion text format by default.

Reply:
Make sense. Should we add unit tests for factories?

Reply:
Currently the factories are integrated with the ObjectMapper and we can verify the functionalities of factories by testing the behaviors of the ObjectMapper. Unit tests for these factories would be writing test code which explicitly states what the production code should do at a syntactic level, i.e. verify `DefaultIonWriterFactory` is returning `IIonWriter`. And unit tests won't help us find bug like this, i.e. the `DefaultIonWriterFactory` is working properly but we are not integrating it correctly in ObjectMapper.

To prevent such bug in the future, we may add more tests to verify each option in `IonSerializationOption` and the combination of these options are working as expected, but that will need exponential number of tests to be exhaustive.

Reply:
Okay. I'm talking about the syntactic level tests actually. Seems to me it's wrong not to have unit tests for public classes.

Reply:
The current implementations of the factories are simple enough that the cost of writing tests outweighs the value of the tests. We may consider to add unit tests to them in the future when they become more complex.